docs/DeveloperGuide.md

```diff
@@ -1,31 +1,400 @@
+On this desktop application, CS2113/T students can attempt bite-sized quizzes, through the **Command Line Interface (CLI)**, to test their understanding of the concepts taught, and serves to consolidate key concepts for easy revision.
 
-### Value proposition
+E-Duke-8, comprises of a Logic component, UI component, Storage component, and Model component. Each component comprises of multiple classes that work in tandem, to fulfil the purpose of our program. 
 
-{Describe the value proposition: what problem does it solve?}
+The purpose of this developer guide is to allow any interested contributors, who wish to develop this learning companion further, understand the inner workings of the program. 
+This understanding will enable such contributors to add value to the current code, by improving its performance, level of interaction or capabilities. 
 
-## User Stories
+### 1.2. Setting up
+
+First, fork this repo, and clone the fork into your computer.
+
+You are recommended to use Intellij IDEA to edit the program.
+
+1. **Configure the JDK**: Ensure Intellij is configured to use JDK 11.
+2. **Import the project as a Gradle project**: Choose the option to import the project as a Gradle project when prompted.
+3. **Verify the setup**: Enter some commands to ensure E-Duke-8 functions as expected. Refer to our [User Guide](https://ay2021s1-cs2113t-f12-3.github.io/tp/UserGuide.html) for more information.
+
+## 2. Design & implementation
+
+### 2.1. Architecture
+
+The high-level design of our program is based on a 3-tier architecture which consists of the Presentation, Application and Database layers. The Logic component, UI component, Storage component, and Model component can be found in each layer as shown in the architecture diagram below.
+
+![Architecture](images/Architecture.png)
+
+### 2.2. Model Component
+
+####  2.2.1. Design of TopicList
+
+1. Listing topics in TopicList
+2. Finding a topic in TopicList
+3. Adding a new note
+4. Deleting an existing note
+5. Listing out all notes in a topic
+
+![TopicList_Class_Diagram](./images/TopicListAndNotes.png)
+
+TopicList is an ArrayList of type Displayable, which is one of two interfaces implemented 
+in the code for EDuke8. As such, many of the commands that manipulate the TopicList make 
+use of the package java.util.ArrayList. The TopicList is used to store Topics.
+
+#### 2.2.2. Implementation of TopicList
+
+**Listing topics in TopicList:**
+
+![TopicListSampleSequence](./images/TopicListSampleSequence.png)
+
+This task is performed by the `TopicList.showTopics()` method.
+
+Step 1: The `parseCommand()` method instantiates a `TopicsCommand` object which then calls the 
+        `TopicList.showTopics()` method.
+Step 2: The `TopicList.showTopics()` method then calls the method `Ui.printTopicList()`. The 
+        current `TopicList` is passed into the called method.
+Step 3: The `Ui.printTopicList()` method then prints out the description of each topic in the 
+        `TopicList`. 
```

Comment:
Consider inserting some line spacing for these lines so that they are rendered properly.
![image](https://user-images.githubusercontent.com/47075579/97397657-b91af200-1924-11eb-88ff-66111cb880ad.png)